src/Utilities/Compiler/Options/AggregateCategorizedAnalyzerConfigOptions.cs

```diff
@@ -95,23 +91,17 @@ public T GetOptionValue<T>(string optionName, SyntaxTree? tree, DiagnosticDescri
 
         private bool TryGetOptionValue<T>(string optionName, OptionKind kind, SyntaxTree? tree, DiagnosticDescriptor? rule, TryParseValue<T> tryParseValue, T defaultValue, [MaybeNullWhen(false)] out T value)
         {
+            value = defaultValue;
+
             if (ReferenceEquals(this, Empty))
             {
-                value = defaultValue;
                 return false;
             }
 
-            // Prefer additional file based options for back compat.
```

Comment:
This comment is the only thing that worries me about this change.
@mavasani Can you provide more details on this? How this is a back compat concern if the compiler should be able to parse the editorconfig correctly whether it's provided as an additional file or not?

Reply:
@Youssef1313 This repo added support for parsing and respecting editorconfig options even prior to compiler adding support in VS2019 16.3. This was done via passing editorconfig files as additional files, as msbuild/compiler did not implicitly discover or respect editorconfig in build. Your change will break anyone on a VS2019 build prior to 16.3 OR if they had editorconfig outside the project directory cone and passed it as an additional file. We can recommend the users to switch to latest VS2019 builds + move from additional file based editorconfig to global config files, so the breaking change has a path forward towards resolution. However, it will be a breaking change indeed.

Reply:
@mavasani Thanks for clarifying. So this suggests that a doc update will be needed to clarify this breaking change?

cc: @gewarren

Reply:
Please file a breaking change issue at https://github.com/dotnet/docs/issues/new?assignees=&labels=&template=dotnet-breaking-change.md&title= once this is merged.

Reply:
I don't expect the original approach for passing editorconfig as additional file would be used that widely, so I feel like we can take the following route:

1. @Youssef1313 - alter your PR slightly so that if we encounter an editorconfig as an additional file, instead of silently ignoring it, we produce a graceful diagnostic message that helps in migration. Implementation wise, you can throw an exception with a clear message that users no longer need to pass editorconfig as an additional file. Either it will be implicitly discovered (if the file is in the project's directory or parent) OR it should be converted into a global config - also recommend moving to latest Dev16 update. The exception message can also point to https://docs.microsoft.com/dotnet/fundamentals/code-analysis/configuration-files. Users will see a special AD0001 analyzer diagnostic with this exception message when they move to this new analyzer package.
2. If someone files an issue on either the analyzers repo or doc repo about them encountering this, then we can consider adding a section at the end of the above docs link about this breaking change and the migration steps. I have a feeling that it might not be required.

Thoughts?

Reply:
Ping @Youssef1313

Reply:
@mavasani I already did that in 4753fd32fe94a1441a76c63ea3cf1e7732f804aa

Reply:
Ah sorry, I forgot. Let me take another final look at the PR.